hail/src/test/scala/is/hail/expr/ir/ExtractIntervalFiltersSuite.scala

```diff
@@ -190,6 +190,24 @@ class ExtractIntervalFiltersSuite extends SparkSuite {
     }
   }
 
+  @Test def testIntervalListFold() {
+    val inIntervals = FastIndexedSeq(
+      Interval(IntervalEndpoint(0, -1), IntervalEndpoint(10, -1)),
```

Comment:
Add a test if they're disjoint.

Reply:
This case should be handled by the unit test for `Interval.union`, I think.

I made the test include a null and a single non-trivial example of a union, since these test that we're getting to the correct (tested) Interval.union functionality.